testing/test_hookrelay.py

```diff
@@ -76,3 +76,32 @@ def hello(self, arg):
     pm.register(Plugin())
     res = pm.hook.hello(arg=3)
     assert res == 4
+
+
+def test_defaults(pm):
+    """Verify that default keyword arguments can be declared on both specs
+    and impls. The default value look up precedence is up as follows:
+        - caller provided value
+        - hookspec default
+        - hookimpl default
+    """
+    class Api:
+        @hookspec
+        def myhook(self, arg, kwarg="default"):
+            "A spec with a default"
+
+    class Plugin:
+        @hookimpl
+        def myhook(self, arg, kwarg="my default"):
+            return kwarg
+
+    pm.register(Plugin())
+
+    # with no spec registered
+    assert pm.hook.myhook(arg='yeah!')[0] == "my default"
```

Comment:
Hmm I'm surprised by this behavior; I would not expect a hook caller to be able to make a call without a registered spec. Is this the bug @hpk42 commented [here](https://github.com/pytest-dev/pluggy/issues/15#issuecomment-261485446)?